nodebuilder/share/module.go

```diff
@@ -34,7 +34,75 @@ func ConstructModule(tp node.Type, cfg *Config, options ...fx.Option) fx.Option
 		fx.Supply(*cfg),
 		fx.Error(cfgErr),
 		fx.Options(options...),
-		fx.Provide(newModule),
+		fx.Provide(newShareModule),
+		shrexSubComponents(),
+	)
+
+	bridgeAndFullComponents := fx.Options(
+		fx.Provide(getters.NewStoreGetter),
+		discoveryComponents(cfg),
+		shrexServerComponents(cfg),
+		edsStoreComponents(cfg),
+		fullAvailabilityComponents(),
+		shrexGetterComponents(cfg),
+		fx.Provide(func(shrexSub *shrexsub.PubSub) shrexsub.BroadcastFn {
+			return shrexSub.Broadcast
+		}),
+	)
+
+	switch tp {
+	case node.Bridge:
+		return fx.Module(
+			"share",
+			baseComponents,
+			bridgeAndFullComponents,
+			fx.Provide(func() peers.Parameters {
+				return cfg.PeerManagerParams
+			}),
+			fx.Provide(peers.NewManager),
```

Comment:
What do you think to pass in bool var `useShrexSub` into  `peerManagerComponents` to allow to use same constructor function for BN and FN/LN but with `false` value?